File: backend/src/main/java/edu/cit/amihan/medibook/doctor/entity/Doctor.java
package edu.cit.amihan.medibook.doctor.entity;

import edu.cit.amihan.medibook.user.entity.User;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "doctors")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Doctor {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "doctor_id")
    private Long doctorId;

    @OneToOne
    @JoinColumn(name = "user_id", nullable = false, unique = true)
    private User user;

    @Column(name = "full_name", nullable = false)
    private String fullName;

    @Column(name = "specialization")
    private String specialization;

    @Column(name = "contact_number")
    private String contactNumber;
}

File: mobile/app/src/main/java/edu/cit/amihan/medibook/feature/doctor/model/Doctor.kt
package edu.cit.amihan.medibook.feature.doctor.model

import com.google.gson.annotations.SerializedName

data class Doctor(
    @SerializedName("doctorId") val doctorId: Long = 0L,
    @SerializedName("fullName") val fullName: String? = null,
    @SerializedName("specialization") val specialization: String? = null,
    @SerializedName("contactNumber") val contactNumber: String? = null,
    @SerializedName("email") val email: String? = null
)